src/components/Editor/ConditionalPanel.js

```diff
@@ -67,21 +72,20 @@ export class ConditionalPanel extends PureComponent<Props> {
   }
 
   clearConditionalPanel() {
-    if (this.cbPanel) {
-      this.cbPanel.clear();
-      this.cbPanel = null;
-    }
-    if (this.scrollParent) {
-      this.scrollParent.removeEventListener("scroll", this.repositionOnScroll);
+    const { editor } = this.props;
+    if (this.expressionOverlay !== null) {
+      editor.monaco.removeOverlayWidget(this.expressionOverlay);
+      this.expressionOverlay = null;
     }
-  }
+    if (this.expressionViewZone >= 0) {
+      editor.monaco.changeViewZones(accessor => {
+        accessor.removeZone(this.expressionViewZone);
+        this.expressionViewZone = -1;
```

Comment:
what's going on here?

Reply:
The way we draw the conditional panel in Monaco is 

* Render a whitespace area after Line A
* Put a DOM node, which is the conditional panel on top of the whitespace area (so in our case, the DOM node is managed by React)


When we hide the conditional panel, we need to remove the whitespace area from the editor. That's what the code is doing here.